acceptance/analyzer_test.go

```diff
@@ -0,0 +1,503 @@
+// +build acceptance
+
+package acceptance
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	ih "github.com/buildpacks/imgutil/testhelpers"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/sclevine/spec"
+	"github.com/sclevine/spec/report"
+
+	"github.com/buildpacks/lifecycle"
+	"github.com/buildpacks/lifecycle/auth"
+	h "github.com/buildpacks/lifecycle/testhelpers"
+)
+
+var (
+	analyzeDockerContext = filepath.Join("testdata", "analyzer", "analyze-image")
+	analyzerBinaryDir    = filepath.Join("testdata", "analyzer", "analyze-image", "container", "cnb", "lifecycle")
+	analyzeImage         = "lifecycle/acceptance/analyzer"
+	analyzerPath         = "/cnb/lifecycle/analyzer"
+	cacheImageRepo       = "lifecycle/acceptance/analyzer/cache-image"
+)
+
+var (
+	registry *ih.DockerRegistry
+)
+
+func TestAnalyzer(t *testing.T) {
+	dockerConfigDir, err := ioutil.TempDir("", "test.docker.config.dir")
+	h.AssertNil(t, err)
+	defer os.RemoveAll(dockerConfigDir)
+
+	registry = ih.NewDockerRegistryWithAuth(dockerConfigDir)
+	registry.Start(t)
+	defer registry.Stop(t)
+
+	os.Setenv("DOCKER_CONFIG", registry.DockerDirectory)
+
+	outDir, err := filepath.Abs(filepath.Join("..", "out"))
+	h.AssertNil(t, err)
+
+	info, err := h.DockerCli(t).Info(context.TODO())
+	h.AssertNil(t, err)
+	osType := info.OSType
+
+	h.ConditionallyBuildLifecycle(t, outDir, osType)
+	h.CopyLifecycle(t, filepath.Join(outDir, osType, "lifecycle"), analyzerBinaryDir)
+
+	buildTestImage(t, analyzeImage, analyzeDockerContext)
+	defer removeTestImage(t, analyzeImage)
+	spec.Run(t, "acceptance", testAnalyzer, spec.Parallel(), spec.Report(report.Terminal{}))
+}
+
+func testAnalyzer(t *testing.T, when spec.G, it spec.S) {
```

Comment:
Following a discussion that we had, I think we should add a test that checks a "good path" of writing the `analyzedMD` output of `analyze` (in `cmd/lifecycle/analyzer.go`).
Currently we only check the content of `analyzedMD` in the unit tests and the creation of this file. Probably it's worth checking its content at least once.